test/Common/HttpApiHandlerTests.cs

```diff
@@ -836,6 +836,140 @@ public async Task StartNewInstance_Returns_HTTP_400_On_Missing_Function()
             Assert.Equal(exceptionMessage, error["ExceptionMessage"].ToString());
         }
 
+        [Theory]
+        [InlineData(false, true)]
+        [InlineData(false, false)]
+        [InlineData(true, true)]
+        [InlineData(true, false)]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public async Task GetEntity_Returns_State_Or_HTTP_404(bool hasKey, bool exists)
+        {
+            string entity = "SomeEntity";
+            string key = hasKey ? Guid.NewGuid().ToString("N") : "";
+            var uriBuilder = new UriBuilder(TestConstants.NotificationUrl);
+
+            uriBuilder.Path += $"/entities/{entity}";
+
+            if (!string.IsNullOrEmpty(key))
+            {
+                uriBuilder.Path += $"/{key}";
+            }
+
+            var testRequest = new HttpRequestMessage
+            {
+                Method = HttpMethod.Get,
+                RequestUri = uriBuilder.Uri,
+            };
+
+            var entityId = new EntityId(entity, key);
+            var result = new EntityStateResponse<JToken>() { EntityExists = exists, EntityState = exists ? new JObject() : null };
+            var clientMock = new Mock<IDurableOrchestrationClient>(MockBehavior.Strict);
+
+            clientMock
+                    .Setup(x => x.ReadEntityStateAsync<JToken>(entityId, null, null))
+                    .Returns(Task.FromResult(result));
+
+            var httpApiHandler = new ExtendedHttpApiHandler(clientMock.Object);
+            var actualResponse = await httpApiHandler.HandleRequestAsync(testRequest);
+
+            if (exists)
+            {
+                Assert.Equal(HttpStatusCode.OK, actualResponse.StatusCode);
+
+                var content = await actualResponse.Content.ReadAsStringAsync();
+                Assert.Equal("{}", content);
+            }
+            else
+            {
+                Assert.Equal(HttpStatusCode.NotFound, actualResponse.StatusCode);
+            }
+        }
+
+        [Theory]
+        [InlineData(false, false, false)]
+        [InlineData(false, false, true, true)]
+        [InlineData(false, false, true, false)]
+        [InlineData(false, true, false)]
+        [InlineData(false, true, true, true)]
+        [InlineData(false, true, true, false)]
+        [InlineData(true, false, false)]
+        [InlineData(true, false, true, true)]
+        [InlineData(true, false, true, false)]
+        [InlineData(true, true, false)]
+        [InlineData(true, true, true, true)]
+        [InlineData(true, true, true, false)]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public async Task SignalEntity_Is_Success(bool hasKey, bool hasOp, bool hasContent, bool hasJsonContent = false)

```

Comment:
Can we add a separate test (or separate condition on this test) for when an entity doesn't exist?

Reply:
The behavior of a signal is identical (whether the entity exists or not) so I don't think we need a separate test.